op-chain-ops/script/script.go

```diff
@@ -86,12 +114,26 @@ type Host struct {
 	srcMaps map[common.Address]*srcmap.SourceMap
 
 	onLabel []func(name string, addr common.Address)
+
+	hooks *Hooks
+}
+
+type Hooks struct {
+	OnBroadcast func(broadcast Broadcast)
+}
+
+var defaultHooks = &Hooks{
+	OnBroadcast: func(broadcast Broadcast) {},
+}
+
+func NewHost(logger log.Logger, fs *foundry.ArtifactsFS, srcFS *foundry.SourceMapFS, executionContext Context) *Host {
+	return NewHostWithHooks(logger, fs, srcFS, executionContext, nil)
 }
 
-// NewHost creates a Host that can load contracts from the given Artifacts FS,
+// NewHostWithHooks creates a Host that can load contracts from the given Artifacts FS,
 // and with an EVM initialized to the given executionContext.
 // Optionally src-map loading may be enabled, by providing a non-nil srcFS to read sources from.
-func NewHost(logger log.Logger, fs *foundry.ArtifactsFS, srcFS *foundry.SourceMapFS, executionContext Context) *Host {
+func NewHostWithHooks(logger log.Logger, fs *foundry.ArtifactsFS, srcFS *foundry.SourceMapFS, executionContext Context, hooks *Hooks) *Host {
```

Comment:
We should either introduce an `Option` pattern to configure `Host`, or keep it as-is, and instantiate host first, then attach optional extras to it with a method. The argument list will keep growing and breaking existing usages otherwise.

Reply:
Cool, I'll just do an option pattern now then. You're referring to functional options in this case, right?